Support new z3 AST node (#13009)
ethereum/solidity: docs/smtchecker.rst

  - if a dynamic ``z3`` library of version 4.8.x is installed in a Linux system (from Solidity 0.7.6);
  - statically in ``soljson.js`` (from Solidity 0.6.9), that is, the Javascript binary of the compiler.

.. note::

>   z3 version 4.8.15 broke ABI compatibility with previous versions and can only

Reviewer: ```suggestion
  z3 version 4.8.15 broke ABI compatibility with previous versions and cannot
```
:-)?

Author: done
